File: result/translate/starchat/python2java/java/COUNT_SORTED_ROWS_MATRIX.java
public static int f_gold ( int[][] mat, int r, int c ) {
        int result = 0;
        for ( int i = 0; i < r; i++ ) {
            int j = 0;
            while ( j < c - 1 && mat[i][j + 1] <= mat[i][j] ) {
                j++;
            }
            if ( j == c - 1 ) {
                result++;
            }
            for ( j = c - 1; j > 0; j-- ) {
                if ( mat[i][j - 1] <= mat[i][j] ) {
                    break;
                }
            }
            if ( j == 0 && c > 1 ) {
                result++;
            }
        }
        return result;
    }

File: result/translate/starchat/java2python/python/COUNT_SORTED_ROWS_MATRIX.py
def f_gold(mat, r, c):
        result = 0
        for i in range(r):
            j = 0
            while j < c - 1:
                if mat[i][j + 1] <= mat[i][j]:
                    break
                j += 1
            if j == c - 1:
                result += 1
        for i in range(r):
            j = c - 1
            while j > 0:
                if mat[i][j - 1] <= mat[i][j]:
                    break
                j -= 1
            if c > 1 and j == 0:
                result += 1
        return result